ams/ams-server/src/main/java/com/netease/arctic/ams/server/optimize/SupportHiveCommit.java

```diff
@@ -0,0 +1,125 @@
+import com.netease.arctic.table.ArcticTable;
+import com.netease.arctic.utils.FileUtil;
+import com.netease.arctic.utils.SerializationUtil;
+import org.apache.commons.collections.CollectionUtils;
+import org.apache.iceberg.DataFile;
+import org.apache.iceberg.StructLike;
+import org.apache.iceberg.types.Types;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.nio.ByteBuffer;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.function.Consumer;
+import java.util.stream.Collectors;
+
+public class SupportHiveCommit extends BaseOptimizeCommit {
+  private static final Logger LOG = LoggerFactory.getLogger(SupportHiveCommit.class);
+
+  protected Consumer<OptimizeTaskItem> updateTargetFiles;
+
+  public SupportHiveCommit(ArcticTable arcticTable,
+                           Map<String, List<OptimizeTaskItem>> optimizeTasksToCommit,
+                           Consumer<OptimizeTaskItem> updateTargetFiles) {
+    super(arcticTable, optimizeTasksToCommit);
+    Preconditions.checkArgument(HiveTableUtil.isHive(arcticTable), "The table not support hive");
+    this.updateTargetFiles = updateTargetFiles;
+  }
+
+  @Override
+  public long commit(TableOptimizeRuntime tableOptimizeRuntime) throws Exception {
+    HMSClient hiveClient = ((SupportHive) arcticTable).getHMSClient();
+    Map<String, String> partitionPathMap = new HashMap<>();
+    Types.StructType partitionSchema = arcticTable.isUnkeyedTable() ?
+        arcticTable.asUnkeyedTable().spec().partitionType() :
+        arcticTable.asKeyedTable().baseTable().spec().partitionType();
+
+    optimizeTasksToCommit.forEach((partition, optimizeTaskItems) -> {
+      // if major optimize task don't contain pos-delete files in a partition, can rewrite or move data files
+      // to hive location
+      if (isPartitionMajorOptimizeSupportHive(optimizeTaskItems)) {
+        for (OptimizeTaskItem optimizeTaskItem : optimizeTaskItems) {
+          BaseOptimizeTaskRuntime optimizeRuntime = optimizeTaskItem.getOptimizeRuntime();
+          List<DataFile> targetFiles = optimizeRuntime.getTargetFiles().stream()
+              .map(fileByte -> (DataFile) SerializationUtil.toInternalTableFile(fileByte))
+              .collect(Collectors.toList());
+
+          List<ByteBuffer> newTargetFiles = new ArrayList<>(targetFiles.size());
+          for (DataFile targetFile : targetFiles) {
+            if (partitionPathMap.get(partition) == null) {
+              List<String> partitionValues =
+                  HivePartitionUtil.partitionValuesAsList(targetFile.partition(), partitionSchema);
+              String partitionPath;
+              if (arcticTable.spec().isUnpartitioned()) {
+                partitionPath = ((SupportHive) arcticTable).hiveLocation();
+              } else {
+                partitionPath =
+                    ((SupportHive) arcticTable).hiveLocation() +
```

Comment:
use the general method in HiveTableUtil to generate new hive location, after PR merged https://github.com/NetEase/arctic/pull/253